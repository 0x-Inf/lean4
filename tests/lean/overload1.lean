variable N : Type
variable f : Bool -> Bool -> Bool
variable g : N -> N -> N
infixl 10 ++ : f
infixl 10 ++ : g
print true ++ false ++ true
set::option lean::pp::notation false
print true ++ false ++ true
variable a : N
variable b : N
print a ++ b ++ a
print true ++ false ++ false