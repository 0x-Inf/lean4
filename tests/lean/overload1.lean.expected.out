  Assumed: N
  Assumed: f
  Assumed: g
⊤ ++ ⊥ ++ ⊤
  Set: lean::pp::notation
f (f true false) true
  Assumed: a
  Assumed: b
g (g a b) a
f (f true false) false
